scripts/Muon/GUI/Common/grouping_tab_widget/grouping_tab_widget_model.py

```diff
@@ -51,17 +51,36 @@ def groups(self):
     def pairs(self):
         return self._groups_and_pairs.pairs
 
+    @property
+    def diffs(self):
+        return self._groups_and_pairs.diffs
+
     @property
     def group_names(self):
         return self._groups_and_pairs.group_names
 
+    @property
+    def diff_names(self):
+        return self._groups_and_pairs.diff_names
+
     @property
     def pair_names(self):
         return self._groups_and_pairs.pair_names
 
+    def get_diffs(self, group_or_pair):
+        return self._groups_and_pairs.get_diffs(group_or_pair)
+
+    def get_names(self, group_or_pair):
+        if group_or_pair == "group":
+            return self.group_names
+        elif group_or_pair == "pair":
+            return self.pair_names
+        else:
+            return []
+
     @property
     def group_and_pair_names(self):
```

Comment:
This is now group pair and diff names,could this be renamed?

Reply:
The units dont change by doing a difference, so I have continued to think of differences as either groups or pairs.